src/components/DetailsMetrics/DetailsMetrics.js

```diff
@@ -139,24 +253,41 @@ const DetailsMetrics = ({ selectedItem }) => {
         params.end = detailsStore.dates.value[1].getTime()
       }
 
-      selectedMetrics.forEach(metric => {
+      ;[...invocationMetric, ...selectedMetrics].forEach(metric => {
         params.name.push(metric.full_name)
       })
 
-      metricsValuesAbortController.current = new AbortController()
+      const newRange = getDateRangeBefore(params)
 
-      setTimeout(() => {
-        dispatch(
-          detailsActions.fetchModelEndpointMetricsValues(
-            selectedItem.metadata.project,
-            selectedItem.metadata.uid,
-            params,
-            metricsValuesAbortController.current.signal
-          )
-        ).then(metricsList => {
-          setMetrics(metricsList)
-        })
-      })
+      const params2 = { name: [] }
+      params2.start = newRange.start
+      params2.end = newRange.end
+      const [{ full_name }] = detailsStore.metricsOptions.all.filter(
+        metric => metric.app === 'mlrun-infra'
+      )
+      params2.name.push(full_name)
+
+      fetchData(params, params2, selectedItem).then()
+    } else if (detailsStore.metricsOptions.all.length !== 0) {
+      const params1 = { name: [] }
+      params1.start =
+        detailsStore.dates?.value[0] === '' ? '' : detailsStore.dates?.value[0].getTime()
+      params1.end =
+        detailsStore.dates?.value[0] === '' ? '' : detailsStore.dates?.value[1].getTime()
+
+      const newRange = getDateRangeBefore(params1)
+      const params2 = { name: [] }
+
+      params2.start = newRange.start
+      params2.end = newRange.end
+
+      const [{ full_name }] = detailsStore.metricsOptions.all.filter(
+        metric => metric.app === 'mlrun-infra'
+      )
+      params1.name.push(full_name)
+      params2.name.push(full_name)
+
+      fetchData(params1, params2, selectedItem).then()
```

Comment:
What is the empty `then`?

Reply:
![Uploading Screenshot 2024-06-05 at 2.16.47 p.m..png…]()